File: 프로그래머스/lv0/120840. 구슬을 나누는 경우의 수/구슬을 나누는 경우의 수.py
from itertools import combinations
def factorial(a):
    result = 1
    for item in range(1, a+1, 1):
        result *= item      #result = result * item
    return result
def solution(balls, share):
    answer = 0
    
    nFact = factorial(balls)
    n_mFact = factorial(balls - share)
    mFact = factorial(share)
    
    
    # items = [i for i in range(balls)]
    # res = list(combinations(items, share))
    
    return nFact // (n_mFact * mFact)
    

File: 프로그래머스/lv0/120840. 구슬을 나누는 경우의 수/구슬을 나누는 경우의 수.js
function solution(balls, share) {
    return Math.round(factorial(balls) / (factorial(balls - share) * factorial(share)));
    
    
}

function fac(number) {
    if(number === 0) return 1;
    return number * fac(number - 1)
}

function factorial(n) {
    res = 1
    for(let i =1;i<=n;i++) {
        res = res * i
    }
    return res
}